app/src/main/java/ip/team13/petowner/ui/activities/ActivitiesFragment.kt

```diff
@@ -2,32 +2,60 @@ package ip.team13.petowner.ui.activities
 
 import android.os.Bundle
 import android.view.View
+import androidx.fragment.app.setFragmentResultListener
 import androidx.navigation.fragment.findNavController
 import ip.team13.petowner.R
 import ip.team13.petowner.core.BaseFragment
+import ip.team13.petowner.core.helpers.AppConstants.ARG_OBJECT
+import ip.team13.petowner.data.domain.ActivityData
+import ip.team13.petowner.data.dto.ActivityEntryModel
 import ip.team13.petowner.databinding.ActivitiesScreenBinding
 import ip.team13.petowner.ui.activities.list.ActivityAdapter
 import org.koin.androidx.viewmodel.ext.android.viewModel
 import org.koin.core.parameter.parametersOf
 
 class ActivitiesFragment : BaseFragment<ActivitiesScreenBinding>() {
 
+    companion object {
+        const val REQUEST_KEY_NEW_ACTIVITY = "REQUEST_KEY_NEW_ACTIVITY"
+    }
+
     override val layout: Int
         get() = R.layout.activities_screen
 
+    private var adapter: ActivityAdapter? = null
+
     private val onAddActivity = {
         findNavController().navigate(R.id.action_activitiesFragment_to_activityDetailsFragment)
     }
 
-    override val viewModel: ActivitiesViewModel by viewModel { parametersOf(onAddActivity) }
+    private val notifyDataSetChange: ((ArrayList<ActivityData>) -> Unit) = {
```

Comment:
We should not bring any data in the View layer as it is against the MVVM architecture. Please check the implementation in `HomeLeaderboardTabViewModel`. It should be mostly the same situation: a list of `@get:Bindable` items sits in the VM, and it is used in the layout with the `app:items` binding adapter. The layout is automatically notified whenever the list contents change.